src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -148,11 +148,14 @@ public void Abort(ConnectionAbortedException ex)
 
         public void Tick()
         {
-            if (_aborted)
+            lock (_protocolSelectionLock)
             {
-                // It's safe to check for timeouts on a dead connection,
-                // but try not to in order to avoid extraneous logs.
-                return;
+                if (_aborted)
+                {
+                    // It's safe to check for timeouts on a dead connection,
+                    // but try not to in order to avoid extraneous logs.
```

Comment:
Why put this lock here? As the comment says, we're just checking `_aborted` to short-circuit as an optimization, not for correctness.

Reply:
Sorry, I ignored it, I changed it back to what it was.